scripts/code_generation/commit_info_gen/src/generate_commit_info.py

```diff
@@ -0,0 +1,74 @@
+import os
+import string
+from argparse import ArgumentParser
+import jinja2
+import git
+
+# dummy data in case of error
+DUMMY_DATA = {
+  "hash": "0000000",
+  "clean": "0"
+}
+
+# validates data dict
+# data["hash"]: 7 char string with short commit hash
+# data["clean"]: "0" if dirty, "1" is clean
+def validateData(data):
+  validLength = len(data["hash"]) == 7
+  validHex = all([c in string.hexdigits for c in data["hash"]])
+  validClean = data["clean"] == "false" or data["clean"] == "true"
+  return validLength and validHex and validClean
```

Comment:
nit: Python convention is to use `snake_case` for variables and function names, and `PascalCase` for classes

Reply:
good catch, fixed!